src/core.js

```diff
@@ -42,8 +42,8 @@ register(require('./traces/scatter'));
 
 // register all registrable components modules
 register([
-    require('./components/fx'),
     require('./components/legend'),
+    require('./components/fx'),
```

Comment:
Heh that was easy 😅 A little fragile though. Can we add a comment at least, saying that fx needs to come after legend for `supplyDefaults` to work correctly? I wonder if there are any other couplings like this?

Reply:
Done in https://github.com/plotly/plotly.js/pull/4687/commits/0d406a02fd927cbba70cd3b9a5be58b265cae3cc

It's a bit fragile although the `hoverlable_test` would fail if the order was changed!